lib/eslint/eslint.js

```diff
@@ -919,8 +920,12 @@ class ESLint {
                         }
 
                         return result;
-                    });
+                    }).catch(error => {
+                        onFatalError(error);
 
+                        // eslint-disable-next-line n/no-process-exit -- Intentionally exit to avoid running the process further.
+                        process.exit();
```

Comment:
We can't use `process.exit()` because it doesn't flush the output buffer before exiting. (We tried this in the past.)

Even so, we absolutely should not use `process.exit()` in this file because it's part of the public API and people won't expect it.

Reply:
You're correct, let me relook at this again.